### DISCUSSION TITLE
An open-source playground that supports Civet

### DISCUSSION THREAD

**Discussion by hatemhosny** [DanielXMoore/Civet#776] [Show and tell]
Thank you for such a nice project.

Please allow me to introduce [LiveCodes](https://livecodes.io), a [feature-rich](https://livecodes.io/docs/features/), [open-source](https://github.com/live-codes/livecodes), [client-side](https://livecodes.io/docs/why/#client-side), code playground that supports [80+ languages and frameworks](https://livecodes.io/docs/why/#language-support). The [announcement post](https://blog.livecodes.io/introducing-livecodes/) gives a general overview.

**Civet** is one of the supported languages.

Civet starter template is available for a quick start:  https://livecodes.io/?template=civet  
(see all [starter templates](https://livecodes.io/?new)).

You can start an empty Civet project using this link: https://livecodes.io/?civet

There is a [compiled code viewer](https://livecodes.io/docs/features/compiled-code) (that shows compiled JS code), [integrated console](https://livecodes.io/docs/features/console) and [test runner](https://livecodes.io/docs/features/tests). [Modules can be imported](https://livecodes.io/docs/features/module-resolution) from npm, denoland, GitHub and others.

Projects can be [shared](https://livecodes.io/docs/features/share/), [exported](https://livecodes.io/docs/features/export/) and [deployed](https://livecodes.io/docs/features/deploy/) (to GitHub Pages), in addition to a lot more [features](https://livecodes.io/docs/features/).

There is even a free [AI code assistant](https://livecodes.io/docs/features/ai) (similar to GitHub Copilot).

Playgrounds can be [embedded](https://livecodes.io/docs/features/embeds) in any webpage, using a powerful, yet easy-to-use, [SDK](https://livecodes.io/docs/sdk).

This is an example for an embedded Civet playground:

```html
<div id="container"></div>

<script type="module">
  import { createPlayground } from 'https://unpkg.com/livecodes';

  createPlayground('#container', {
    params: {
      civet: "console.log 'Hello Civet!'",
      console: "open",
    },
  });
</script>
```
[Preview](https://livecodes.io/?mode=result&x=code/N4IgLglmA2CmIC4QGEIDdZgAQBFYFsB7EAGhABNYBnAYwCcIAHSQgO0RFPAEMBzKxAG0AumW41IGAKLkohOh3zc6AawCujLktUbEoaN1a81feEgAWYfNC402YWKzAcAPLLRYI5ALwAdEHZO3BCssHT+AHwuAPTuEb7hrAm+rC60DMxYYACejLB+IETkanCRyaxYnviM8tjAWPSw3A4ACgbZvHSEaqzkWAC+WABmXfhYAOSWYIxUCNHRPYwqvAB0dvjR0OiwdpRU4wDc5eWVjc2wbdwdXT3kABTjAMSBYMGhdOMkWMAnlViMym4+Fm31+fwa2zACCw-kCVEIcBW0EIvAmAAlYNBkVhUBgwABCcb+Ehgv5whGwaH+Qh5JKkUn9EmJSr9ACUR0SMXSTDAERAjJAVBycD0IAMRhMvDMASoAjIL0cziQACNCORsqDmVhtLwQtCAKwABkYAA8OSl+vyyNzmKLxcZTBwAFbcNDcG3OeX2RUcK2C4XUcywTACBAia30Hmh8MyqgtOjUTC+siMLo0ajwujR0QBNRCwj4ADKmEgRlDwAFEGqtXLApyeVrZAcQvLYsMDqlHHr1Ejtq9Th9SD9GCzEDYHAAjPr+UA)

Please refer to the [SDK documentations](https://livecodes.io/docs/sdk) for details.

Comprehensive [documentations](https://livecodes.io/docs/) are available with live demos, code samples and screenshots.

LiveCodes is [open-source](https://github.com/live-codes/livecodes), free with unlimited usage, no ads and no account required. It can be easily [self-hosted](https://livecodes.io/docs/features/self-hosting) (if you want), and can be used for commercial projects ([MIT license](https://livecodes.io/docs/license)).

Disclosure: obviously, I'm the author of LiveCodes.

I thought you would be interested to know about [LiveCodes](https://livecodes.io). Otherwise, please feel free to close this.

**Comment by dux** [comment]
Very nice, thx
